systemtest/src/test/java/io/strimzi/systemtest/dynamicconfiguration/DynamicConfigurationIsolatedST.java

```diff
@@ -0,0 +1,374 @@
+        kafkaConfig.put("transaction.state.log.replication.factor", "1");
+        kafkaConfig.put("default.replication.factor", "1");
+        kafkaConfig.put("log.message.format.version", "2.4");
+
+        Map<String, Object> updatedKafkaConfig = new HashMap<>();
+        updatedKafkaConfig.put("offsets.topic.replication.factor", "1");
+        updatedKafkaConfig.put("transaction.state.log.replication.factor", "1");
+        updatedKafkaConfig.put("default.replication.factor", "1");
+        updatedKafkaConfig.put("log.message.format.version", "2.4");
+        updatedKafkaConfig.put("unclean.leader.election.enable", "true");
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, kafkaReplicas, 1)
+                .editSpec()
+                    .editKafka()
+                        .withConfig(kafkaConfig)
+                    .endKafka()
+                .endSpec()
+                .done();
+
+        String kafkaConfiguration = kubeClient().getConfigMap(KafkaResources.kafkaMetricsAndLogConfigMapName(CLUSTER_NAME)).getData().get("server.config");
+        assertThat(kafkaConfiguration, containsString("offsets.topic.replication.factor=1"));
+        assertThat(kafkaConfiguration, containsString("transaction.state.log.replication.factor=1"));
+        assertThat(kafkaConfiguration, containsString("default.replication.factor=1"));
+        assertThat(kafkaConfiguration, containsString("log.message.format.version=2.4"));
+
+        String kafkaConfigurationFromPod = cmdKubeClient().execInPod(KafkaResources.kafkaPodName(CLUSTER_NAME, 0), "/bin/bash", "-c", "bin/kafka-configs.sh --bootstrap-server localhost:9092 --entity-type brokers --entity-name 0 --describe").out();
+        assertThat(kafkaConfigurationFromPod, is("Dynamic configs for broker 0 are:\n"));
+
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(kafkaStatefulSetName(CLUSTER_NAME));
+
+        LOGGER.info("Updating configuration of Kafka cluster");
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            KafkaClusterSpec kafkaClusterSpec = k.getSpec().getKafka();
+            kafkaClusterSpec.setConfig(updatedKafkaConfig);
+        });
+
+        PodUtils.verifyThatRunningPodsAreStable(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
+        assertThat(StatefulSetUtils.ssHasRolled(kafkaStatefulSetName(CLUSTER_NAME), kafkaPods), is(false));
+
+        LOGGER.info("Verify values after update");
+        kafkaConfiguration = kubeClient().getConfigMap(KafkaResources.kafkaMetricsAndLogConfigMapName(CLUSTER_NAME)).getData().get("server.config");
+        assertThat(kafkaConfiguration, containsString("offsets.topic.replication.factor=1"));
+        assertThat(kafkaConfiguration, containsString("transaction.state.log.replication.factor=1"));
+        assertThat(kafkaConfiguration, containsString("default.replication.factor=1"));
+        assertThat(kafkaConfiguration, containsString("log.message.format.version=2.4"));
```

Comment:
The `unclean.leader.election.enable` should show up here as well. So maybe you can assert it too?